solver/src/driver/solver_settlements.rs

```diff
@@ -27,14 +27,25 @@ pub fn filter_bad_settlements(
 #[derive(Debug)]
 pub struct SolverWithSettlements {
     pub name: &'static str,
-    pub settlements: Vec<RatedSettlement>,
+    pub settlements: Vec<Settlement>,
 }
 
 // Each individual settlement has an objective value.
 #[derive(Debug)]
 pub struct RatedSettlement {
     pub settlement: Settlement,
-    pub objective_value: BigRational,
+    pub surplus: BigRational,
+    pub gas_estimate: BigRational,
+}
+
+impl RatedSettlement {
+    pub fn objective_value(&self, gas_price: f64) -> BigRational {
+        // I can't see why a f64 couldn't be represented as a BigRational. Thoughts?
```

Comment:
IEEE 754 floating point numbers can represent positive or negative infinity, and NaN. Those cannot be represented as BigRationals

Reply:
Right of course. I've used unwrap anyway, since we're not expecting any of those values.